client.go

```diff
@@ -376,13 +381,18 @@ func (c *Client) runOnce() error {
 			return err
 		}
 
+		wg.Add(1) // Confirms consumer.
```

Comment:
nit-picking here, but given the comments and also that we increment the ones for confirm consumer and publisher in this scope but not for the replies consumer which is started in a different function, we might be able to align them?

If we pass these to the go routines, the incrementation isn't arbitrary and thus the comment isn't needed.

Thinking instead of this:

```go
func f() {
    wg := sync.WaitGroup{}

    someFn(&wg) // Adds somewhere else

    wg.Add(1) // Foo
    go foo(&wg)

    wg.add(1) // Bar
    go bar(&wg)
}

func someFn(wg *WaitGroup) {
    wg.Add(1)
    go foo(wg)
}

func foo (wg *WaitGroup) {
    defer wg.Done()
    for {}
}

func bar(wg *WaitGroup) {
    defer wg.Done()
    for {}
}
```

We could increment them where used and close to where they're done?

```go
func f() {
    wg := sync.WaitGroup{}

    someFn(&wg)

    go foo(&wg)
    go bar(&wg)
}

func someFn(wg *WaitGroup) {
    go foo(wg)
}

func foo (wg *WaitGroup) {
    wg.Add(1)
    defer wg.Done()
    for {}
}

func bar(wg *WaitGroup) {
    wg.Add(1)
    defer wg.Done()
    for {}
}
```

Reply:
Yes, I agree. The problem is that doing `wg.Add(1)` before starting the goroutine can race since you aren't allowed to call `wg.Add(1)` after `wg.Wait()`.

Reply:
Right, actually didn't feel right to suggest calling add inside the go routine :) Let's keep it as is!